src/System.Private.ServiceModel/tools/SelfHostedWcfService/Program.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.ServiceModel;
+using System.ServiceModel.Description;
+using System.ServiceModel.Web;
+using WcfService;
+
+namespace SelfHostedWCFService
+{
+    public class SelfHostedWCFService
+    {
+        static int httpPort = 8081;
+        static int httpsPort = 44285;
+        static int tcpPort = 809;
+        static int websocketPort = 8083;
+        static int websocketsPort = 8084;
+
+        static private void GetPorts()
+        {
+            httpPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("httpPort")) ? httpPort : int.Parse(Environment.GetEnvironmentVariable("httpPort"));
+            httpsPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("httpsPort")) ? httpsPort : int.Parse(Environment.GetEnvironmentVariable("httpsPort"));
+            tcpPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("tcpPort")) ? tcpPort : int.Parse(Environment.GetEnvironmentVariable("tcpPort"));
+            websocketPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("websocketPort")) ? websocketPort : int.Parse(Environment.GetEnvironmentVariable("websocketPort"));
+            websocketsPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("websocketsPort")) ? websocketsPort : int.Parse(Environment.GetEnvironmentVariable("websocketsPort"));
```

Comment:
I don't like reinventing a new way to specify and access ports.  TestProperties does this better.  Why didn't you use a known working system for this? Please open an issue to share TestProperties and to parse command line arguments into them (as Bridge.exe does).